wip

aadityasondhi opened the pull request.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/99256)
<!-- Reviewable:end -->

aadityasondhi committed:
roachtest: make `follower-reads` test use protobuf encoded requests

Previously, the roachtest for
`follower-reads/mixed-version/single-region` used JSON encoding for the
`ts/query` endpoint. This caused the client to send the newly updated
`timerseries` proto to an older cluster in JSON format. Although this
was defined as an `optional` field in the proto, since it was encoded in
JSON, that information was last over the wire.

In other places, such as DBConsole, we expect the client to be sending
protobuf encoded messages and not JSON. In the same way, roachtests
(another client) should do the same. When this message is sent as a
protobuf, the `optional` tag of the field is encoded in the message and
the server is able to process is it as such.

There are related links in the issue:
https://github.com/cockroachdb/cockroach/issues/99117.

Fixes #99117.

Release note: None
